core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -288,11 +292,10 @@ class ReplicaFetcherThread(name: String,
         requestMap.put(topicPartition, new JFetchRequest.PartitionData(partitionFetchState.offset, fetchSize))
     }
 
-    val request =
-      if (fetchRequestVersion >= 3) JFetchRequest.fromReplica(replicaId, maxWait, minBytes, maxBytes, requestMap)
-      else JFetchRequest.fromReplica(replicaId, maxWait, minBytes, requestMap)
-
-    new FetchRequest(request)
+    val requestBuilder = new JFetchRequest.Builder(maxWait, minBytes, requestMap).
+        setReplicaId(replicaId).setMaxBytes(maxBytes)
+    requestBuilder.setVersion(fetchRequestVersion)
```

Comment:
setVersion is a method on the base class, AbstractRequest#Builder.  So it returns an instance of AbstractRequest#Builder rather than an instance of the derived class FetchRequest#Builder, which is awkward for Scala's type inference.